### PR TITLE
Add gset support

### PR DESCRIPTION
**Pull request by russelldb**
Rebased bet365 branch on `develop-2.2`. Would appreciate eyes from @binarytemple-bet365, please?

### CODE REVIEW

File: src/riak_kv_crdt.erl [hunk 1]
```diff
@@ -509,7 +516,9 @@ to_record(?MAP_TYPE, Val) ->
 to_record(?SET_TYPE, Val) ->
     ?SET_TYPE(Val);
 to_record(?HLL_TYPE, Val) ->
-    ?HLL_TYPE(Val).
+    ?HLL_TYPE(Val);
```
Comment (binarytemple-external) [inline src/riak_kv_crdt.erl:519]: Correct in the context of this integration - a future refactor might visit the macro definitions though: 

```

-define(COUNTER_TYPE(Val), #crdt{mod=?COUNTER_TYPE, ctype="application/riak_counter", value=Val}).

-define(SET_TYPE(Val), #crdt{mod=?SET_TYPE, ctype="application/riak_set", value=Val}).

-define(HLL_TYPE(Val), #crdt{mod=?HLL_TYPE,
                             ctype="application/riak_hll",
                             value=Val}).

-define(GSET_TYPE(Val), #crdt{mod=?GSET_TYPE, ctype="application/riak_gset", value=Val}).

-define(MAP_TYPE(Val), #crdt{mod=?MAP_TYPE, ctype="application/riak_map", value=Val}).
```

Seems like an extra level of indirection with no apparent benefit.

**Comment by binarytemple-external** [comment]
Looks good to me @russelldb

**Comment by russelldb** [comment]
+1 from me now

Wonder if anyone from KV wants to have a look and a say?
(@JeetKunDoug  you want to suggest someone to cast an eye over it?)

**Comment by russelldb** [comment]
Closing in favour of https://github.com/basho/riak_kv/pull/1523

**Pull request closed without merging**